app/controllers/boxes_controller.rb

```diff
@@ -96,12 +108,38 @@ def load_box
     @box = Box.where(institution: @navigation_context.institution).find(params.fetch(:id))
   end
 
+  def load_box_print
+    begin
+      @box = Box.where(institution: @navigation_context.institution).find(params.fetch(:id))
+    rescue ActiveRecord::RecordNotFound
+      @boxes = TransferPackage
+        .within( @navigation_context.institution )
+        .find( params.fetch( :transfer_package ) )
+        .boxes
+      @box = @boxes.find( params.fetch( :id ) )
+    end
```

Comment:
> Relying on rescue as a fall back is quite inefficient. It would be better to avoid raising `RecordNotFound` in the first place! We can use the non-raising alternative to `find(id)`, which is `where(id: id).take`. If the value of that expression is `nil`, fall back to the other query.

I'm not sure if I understood....you mean like this?
```
@box = Box.where(institution: @navigation_context.institution, id: params.fetch(:id))
    if @box == nil
      @boxes = TransferPackage
        .within( @navigation_context.institution )
        .boxes
      @box = @boxes.find( params.fetch( :id ) )
    end
```

Reply:
Basically, yes. You need an `.take` on the first expression though.

Reply:
I don't know why it doesn't work. That's why I used that `RecordNotFound` solution.
![image](https://user-images.githubusercontent.com/19227697/181832589-ed94bb99-4fbd-4b64-bec7-3cd8936279fd.png)

Reply:
Ok, it's working